api.planx.uk/modules/flows/validate/service.ts

```diff
@@ -55,16 +56,20 @@ const validateAndDiffFlow = async (
   const validationChecks = [];
   const sections = validateSections(flattenedFlow);
   const inviteToPay = validateInviteToPay(flattenedFlow);
-  validationChecks.push(sections, inviteToPay);
+  const fileTypes = validateFileTypes(flattenedFlow);
+  validationChecks.push(sections, inviteToPay, fileTypes);
 
-  // Sort validation checks by status: Fail, Pass, Not applicable
-  const applicableChecks = validationChecks
-    .filter((v) => v.status !== "Not applicable")
-    .sort((a, b) => a.status.localeCompare(b.status));
+  // Arrange list of validation checks in order of status: Fail, Warn, Pass, Not applicable
+  const failingChecks = validationChecks.filter((v) => v.status == "Fail");
+  const warningChecks = validationChecks.filter((v) => v.status === "Warn");
+  const passingChecks = validationChecks.filter((v) => v.status === "Pass");
   const notApplicableChecks = validationChecks.filter(
     (v) => v.status === "Not applicable",
   );
-  const sortedValidationChecks = applicableChecks.concat(notApplicableChecks);
+  const sortedValidationChecks = failingChecks
+    .concat(warningChecks)
+    .concat(passingChecks)
+    .concat(notApplicableChecks);
```

Comment:
Anyone have a more clever suggestion for how to sort in a custom order by "status" ? This is explicit and easy-to-read, but a bit lengthy and I feel like there's probably some cool one-liner?

Reply:
Yeah I can think of a few ways!

`Map.groupBy` is pretty new but seems to have decent coverage https://caniuse.com/?search=map.groupby

I think it would then just be `Map.groupBy(passingChecks, (check) => check.status))`.

If we hit browser compatibility issues here we could import `groupBy` from lodash, or use a `reduce()`.

Docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/groupBy

Reply:
Ah just realised we're in the API here on Node 18 - I don't think `Map.groupBy()` would be available.

Lodash or `reduce()` would be the way to go here 👍